cli/src/main/java/org/wildfly/glow/jbang/JBangIntegration.java

```diff
@@ -0,0 +1,186 @@
+                        while ((bytesRead = is.read(buffer)) != -1) {
+                            os.write(buffer, 0, bytesRead);
+                        }
+                    }
+                }
+            }
+            System.out.println("JAR extracted successfully to " + destDir);
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    public static Path toWar(Path appClasses) throws IOException {
+        Path parent = appClasses.getParent();
+        Path webInf = parent.resolve("WEB-INF");
+        Path webClassesDir = webInf.resolve("classes");
+
+        Files.createDirectories(webClassesDir);
+
+        Files.walkFileTree(appClasses, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                Path targetPath = webClassesDir.resolve(appClasses.relativize(file));
+                Files.createDirectories(targetPath.getParent());
+                Files.copy(file, targetPath, StandardCopyOption.REPLACE_EXISTING);
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                Files.createDirectories(webClassesDir.resolve(appClasses.relativize(dir)));
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+        String warName = parent.getFileName().toString().split("\\.")[0] + ".war";
+        Path warFile = parent.resolve(warName);
+
+        try (ZipOutputStream zipOut = new ZipOutputStream(new FileOutputStream(warFile.toFile()))) {
+            Files.walk(webInf).filter(Files::isRegularFile).forEach(file -> {
+                try {
+                    String zipEntryName = parent.relativize(file).toString().replace("\\", "/");
+                    zipOut.putNextEntry(new ZipEntry(zipEntryName));
+                    Files.copy(file, zipOut);
+                    zipOut.closeEntry();
+                } catch (IOException e) {
+                    // ignore
+                }
+            });
+        }
+
+        return warFile;
+    }
+
+    public static Path runGlowScan(Path glowJar, Path war, Optional<String> glowArgs) throws Exception {
+        System.out.println("Scanning " + war + " with " + glowJar);
+        // todo clean up this temp dir
+        Path temp = Files.createTempDirectory("glow");
+
+        // TODO Find the right jdk version to run Glow in a forked process
+        // with something like `jbang jdk list --format=json`
+        // for now let's use "java"
+
+        List<String> args = new ArrayList<>();
+        args.addAll(List.of("java", "-jar", glowJar.toString(),
+                "scan",
+                "--provision=BOOTABLE_JAR",
+                "--output-dir=" + temp));
+        if (glowArgs.isPresent()) {
+            args.addAll(Arrays.asList(glowArgs.get().split(" ")));
```

Comment:
Should the split be `\s+` so we can line terminates and multiple spaced arguments?

Reply:
You're, right, using `\s+` is better.

`glowArgs` string comes from a comment in the `java` file eg

```
//GLOW --add-ons=health,metrics --server-version=35.0.1.Final
```

I'm not sure we should support line terminates.
One thing we could do however is that for now, I'm taking the first `//GLOW` comment to create the args
but I could change it to take _any_  `//GLOW` comments for more clarity:

```
//GLOW --add-ons=health,metrics 
//GLOW --server-version=35.0.1.Final
//GLOW --spaces=incubating
```

wdyt?

Reply:
fixed in https://github.com/wildfly/wildfly-glow/pull/113/commits/4436201ec23188f30d769322a1233f1e0bd6e60c

Reply:
Ah right, line terminators definitely don't make sense here. Documenting is probably good, but that is a standard JBang way to define things so it might not be critical.